stock_picking_force_assign/models/stock_picking.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import functools
+from odoo import _, exceptions, models, tools
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def action_force_assign_pickings(self):
+        """ Unreserve other pickings in order to reserve pickings in self """
+        link_template = '<a href="#" data-oe-model="%s" data-oe-id="%d">%s</a>'
+        to_unreserve = self._force_assign_find_moves()
+        to_unreserve._do_unreserve()
+        self.message_post(_(
+            'Unreserved picking(s) %s in order to assign this one'
+        ) % ', '.join(to_unreserve.mapped('picking_id').mapped(
+            lambda x: link_template % (x._name, x.id, x.name)
+        )))
+        to_unreserve.mapped('picking_id').message_post(_(
+            'Unreserved this picking in order to assign %s'
+        ) % ', '.join(self.mapped(
+            lambda x: link_template % (x._name, x.id, x.name)
+        )))
+        return self.action_assign()
+
+    def _force_assign_find_moves(self):
+        """ Return moves to unreserve in order to reserve pickings in self """
+        location = self.mapped('location_id')
+        result = self.env['stock.move']
+        assert len(location) == 1, 'Pickings need to be from the same location'
+        float_compare = functools.partial(
+            tools.float_compare,
+            precision_digits=result._fields['product_qty'].digits,
+        )
+
+        for move_line in self.mapped('move_lines'):
```

Comment:
You could consider changing variable name `move_line` to `stock_move` or `move`, as an act of resistance against the terminology fu in core Odoo, but it's just a nice to have.

Reply:
I'm all for the resistance!

Reply:
Alright comrade!